### PR TITLE
OSDOCS-9784: Bump HAProxy version to 2.8

### PR DESCRIPTION
**Pull request by ahardin-rh**
Version(s):
4.16

Issue:
https://issues.redhat.com/browse/OSDOCS-9784

This also relates to https://github.com/openshift/openshift-docs/pull/72835

Link to docs preview:

https://73008--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ingress-operator#nw-http2-haproxy_configuring-ingress

https://73008--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ingress-operator#nw-ingress-converting-http-header-case_configuring-ingress

### CODE REVIEW

**Review by mjpytlak — COMMENTED** [review]
Peer review lgtm, @ahardin-rh

**Comment by ShudiLi** [comment]
/lgtm
thanks

File: modules/nw-http2-haproxy.adoc [hunk 1]
```diff
@@ -16,7 +16,7 @@ The connection from HAProxy to the application pod can use HTTP/2 only for re-en
 
 [WARNING]
 ====
-Using WebSockets with a re-encrypt route and with HTTP/2 enabled on an Ingress Controller requires WebSocket support over HTTP/2. WebSockets over HTTP/2 is a feature of HAProxy 2.6, which is unsupported in {product-title} at this time.
+Using WebSockets with a re-encrypt route and with HTTP/2 enabled on an Ingress Controller requires WebSocket support over HTTP/2. WebSockets over HTTP/2 is a feature of HAProxy 2.8, which is unsupported in {product-title} at this time.
```
Comment (frobware) [inline modules/nw-http2-haproxy.adoc:19]: We need to remove this section. Websockets over HTTP/2 were first supported in haproxy-2.4. With OCP 4.14 we've been running haproxy 2.6, and now in OCP 4.16 the bump to haproxy 2.8 per this PR. When we drop this text we need to backport the doc changes to  OCP 4.15 and OCP 4.14.

Comment (ahardin-rh) [reply modules/nw-http2-haproxy.adoc]: @frobware Thank you for your guidance! Updated. I will backport the removal for 4.15 and 4.14 in separate PRs.

**Pull request merged**